pkg/controller/machinehealthcheck/machinehealthcheck_controller.go

```diff
@@ -238,16 +242,9 @@ func (r *ReconcileMachineHealthCheck) Reconcile(ctx context.Context, request rec
 		klog.Errorf("Reconciling %s: error patching status: %v", request.String(), err)
 		return reconcile.Result{}, err
 	}
-
-	// remediate
-	for _, t := range needRemediationTargets {
-		klog.V(3).Infof("Reconciling %s: meet unhealthy criteria, triggers remediation", t.string())
-		if err := t.remediate(r); err != nil {
-			klog.Errorf("Reconciling %s: error remediating: %v", t.string(), err)
-			errList = append(errList, err)
-		}
-	}
-
+	errList = r.remediate(ctx, needRemediationTargets, errList, mhc)
+	// deletes External Machine Remediation for healthy machines - indicating remediation was successful
+	r.cleanEMR(ctx, currentHealthy, mhc)
```

Comment:
I think it should belong entirely to `externalRemediation` logic block, so the default internal remediation flow is unchanged.

Reply:
I defiantly see your point, but I'm not sure the alternative is better.

The major thing I'm bothered with is that it'll will force us to add `currentHealthy` to the remediate method input params - which does not make a lot of sense to me.

Another minor issue - currently `externalRemediation` and `internalRemediation` takes a single target as input - this change will also cause a refactoring for those method to take the whole slice.

I don't have a strong opinion either way - but I thought it worth bringing these things up before making the change.
Let me know what you think.